workflow/ttlcontroller/ttlcontroller.go

```diff
@@ -236,11 +239,19 @@ func timeLeft(wf *wfv1.Workflow, since *time.Time) (*time.Duration, *time.Time)
 	}
 }
 
-func getTTLStrategy(wf *wfv1.Workflow) *wfv1.TTLStrategy {
-	ttlStrategy := wf.Spec.TTLStrategy
-
-	if ttlStrategy == nil && wf.Status.StoredWorkflowSpec != nil && wf.Status.StoredWorkflowSpec.TTLStrategy != nil {
-		return wf.Status.StoredWorkflowSpec.TTLStrategy
+func getTTLStrategy(wf *wfv1.Workflow, defaultTTLStrategy *wfv1.TTLStrategy) *wfv1.TTLStrategy {
+	var ttlStrategy *wfv1.TTLStrategy
+	// TTLStrategy from Workflow default from Config
+	if defaultTTLStrategy != nil {
+		ttlStrategy = defaultTTLStrategy
+	}
+	// TTLStrategy from WorkflowTemplate
+	if wf.Status.StoredWorkflowSpec != nil && wf.Status.StoredWorkflowSpec.GetTTLStrategy() != nil {
```

Comment:
I think you can create
```golang
func (w *Workflow) GetTTLStrategy(defaultTTLStrategy *wfv1.TTLStrategy) {
  if w == nil {return defaultTTLStrategy}
  if wf.Status.StoredWorkflowSpec != nil && wf.Status.StoredWorkflowSpec.GetTTLStrategy() != nil {
    return wf.Status.StoredWorkflowSpec.GetTTLStrategy()
  }
  if wf.Spec.GetTTLStrategy()!=nil {
    return  wf.Spec.GetTTLStrategy()
  }
  return defaultTTLStrategy
}
```

Reply:
This is `GetTTLStrategy` is only accessed in the TTL controller. I didn't see the benefit to move to the Workflow object level.

Reply:
can you test it easily here? would be easier to test as a object func, I tend to find that be the case